src/index.js

```diff
@@ -100,24 +100,22 @@ class SoshContentScript extends ContentScript {
       throw new Error(`${msg} failed to meet conditions`)
     }
   }
+
   async navigateToLoginForm() {
     this.log('info', '🤖 navigateToLoginForm starts')
+    if (await this.isElementInWorker('#login-label')) {
+      this.log('info', 'Already on loginPage, returning true')
+      return true
+    }
     await this.goto(LOGIN_FORM_PAGE)
-    await this.PromiseRaceWithError(
-      [
-        this.waitForElementInWorker('#login-label'),
-        this.waitForElementInWorker('#password-label'),
-        this.waitForElementInWorker(
-          'button[data-testid="button-keepconnected"]'
-        ),
-        this.waitForElementInWorker('div[class*="captcha_responseContainer"]'),
-        this.waitForElementInWorker('#undefined-label'),
-        this.waitForElementInWorker('#oecs__popin-icon-Identification'),
-        this.waitForElementInWorker('#changeAccountLink'),
-        this.waitForErrorUrl()
-      ],
-      'navigateToLoginForm: waiting for login page load'
-    )
+    await Promise.race([
```

Comment:
Why did you remove the use of PromiseRaceWithError ?

Reply:
It was unvoluntarily, will be fixed 👍